### DISCUSSION TITLE
Cannot set default.timeout-minutes

### DISCUSSION THREAD

**Discussion by jszwedko** [community/community#25472] [Actions]
<p>Looking at <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#defaults" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#defaults</a> it seems like I should be able to define a default <code>timeout-minutes</code> that will be applied to each job, however, when I have:</p>
<pre><code class="lang-auto">defaults:
  timeout-minutes: 15
</code></pre>
<p>In the workflow, it reports that the key was unexpected:</p>
<pre><code class="lang-auto"> Invalid workflow file

The workflow is not valid. .github/workflows/code.yml (Line: 14, Col: 3): Unexpected value 'timeout-minutes'
</code></pre>
<p>Should I be able to set that key there? If not, I think the documentation for <code>defaults</code> could use a little more clarification as to what keys are allowed.</p>

**Comment by ghost** [comment]
<p>It might be a syntax issue – could you try something like this?</p>
<pre><code class="lang-auto">jobs:
  Test:
    - name: Run test
      timeout-minutes: 10
      run: make unit-test
</code></pre>
<p>If that doesn’t work, could you share a copy of your workflow .yml file?</p>

**Comment by jszwedko** [comment]
<p>Thanks for the response! We are able to set the value at the <code>job</code> level, but this requires setting the timeout for every job we add.</p>
<p>The documentation for <code>defaults</code> says:</p>
<blockquote>
<p>A <code>map</code> of default settings that will apply to all jobs in the workflow. You can also set default settings that are only available to a job. For more information, see <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_iddefaults" rel="nofollow noopener"> <code>jobs.&lt;job_id&gt;.defaults</code> </a>.</p>
</blockquote>
<p><a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#defaults" class="onebox" target="_blank" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#defaults</a></p>
<p>Which made me think I could set the <code>timeout-minutes</code> there and have it apply to each job, but this does not appear to be the case. Could you clarify which keys are able to be set in <code>defaults</code>?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<aside class="quote no-group" data-username="jszwedko" data-post="3" data-topic="118164">
<div class="title">
<div class="quote-controls"></div>
 jszwedko:</div>
<blockquote>
<p>Could you clarify which keys are able to be set in <code>defaults</code> ?</p>
</blockquote>
</aside>
<p>Currently, only <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#defaultsrun" rel="nofollow noopener"> <strong><code>defaults.run</code></strong> </a> is supported .<br>
I would encourage you to share your requirement for <code>timeout-minutes</code> in the <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">Feedback form for GitHub Actions</a>.</p>

**Comment by jszwedko** [comment]
<p>Aha, I see, thanks for clarifying. I was thinking that <code>defaults.run</code> was simply expanded documentation for that default but I see that it is the only supported option. I think the documentation could be clearer to that point. I’ll fill out the feedback form.</p>
<p>Thank you!</p>

**Comment by chingc** [comment]
<p><a class="mention" href="/jszwedko">@jszwedko</a> Today I tried</p>
<pre><code class="lang-auto">defaults:
  timeout-minutes: 15
</code></pre>
<p>and ran into the same thing as you did.  It would be nice if that worked.</p>
<p>According to <a class="mention" href="/yanjingzhu">@yanjingzhu</a> it seems like only <code>defaults.run</code> is supported which makes this feature very limited and very specific to certain use cases.  :frowning:</p>
